app/main.py

```diff
@@ -132,6 +132,17 @@ async def delete(request):
     status = await (dqueue.cancel(ids) if where == 'queue' else dqueue.clear(ids))
     return web.Response(text=serializer.encode(status))
 
+@routes.get(config.URL_PREFIX + 'history')
+async def history(request):
+    history = { 'done': [], 'queue': []}

```

Comment:
Maybe you can do it all like this --
```
history = { 'done': [v for k, v in dqueue.queue.saved_items()], 'queue': [v for k, v in dqueue.done.saved_items()]}
```
Can you see if this works? You probably don't even need to call `vars()`. But you may need to switch from `json.dumps` to `serializer.encode()` below.

Reply:
Done